File: 0152-maximum-product-subarray/0152-maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int left = 1;
        int right = 1;
        int maxprod = INT_MIN;
        int n = nums.size();

        for(int i = 0 ; i < n; i++){
            left *= nums[i];
            right *= nums[n-i-1];

            maxprod = max(maxprod , max(left ,right));

            if(left == 0) left = 1;
            if(right == 0) right = 1;
        }
        return maxprod;
    }
};

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        int maxprod = Integer.MIN_VALUE;
        int lprod = 1;
        int rprod = 1;
        int n = nums.length;

        for(int i = 0 ; i < n ; i++){
            lprod *= nums[i];
            rprod *= nums[n-i-1];
            maxprod = Math.max(maxprod , Math.max(lprod , rprod));
            if(lprod == 0) lprod = 1;
            if(rprod == 0) rprod = 1;
        }
        return maxprod;
    }
}